src/expr/node.cpp

```diff
@@ -31,9 +31,21 @@ TypeCheckingExceptionPrivate::TypeCheckingExceptionPrivate(TNode node,
     : Exception(message), d_node(new Node(node))
 {
 #ifdef CVC4_DEBUG
+  std::stringstream ss;
   LastExceptionBuffer* current = LastExceptionBuffer::getCurrent();
   if(current != NULL){
-    current->setContents(toString().c_str());
+    // Since this node is malformed, we cannot use toString().
+    // Instead, we print the kind and the children.
+    ss << message << " "
+       << "node kind: " << node.getKind() << ". children: ";
+    int i = 0;
+    for (const TNode& child : node)
+    {
+      ss << "child " << i << ": " << child << ". ";
+      i++;
+    }
+    string ssstring = ss.str();
+    current->setContents(ssstring.c_str());
```

Comment:
You can use a one-liner here: `current->setContents(ss.str().c_str());`

Reply:
The reason for the split is this discussion at stackoverflow, according to which ss.str() might be deallocated before c_str() is called:

https://stackoverflow.com/questions/21034834/is-there-issue-will-stringstream-str-c-str
https://stackoverflow.com/questions/20883180/why-does-this-work-returning-c-string-literal-from-stdstring-function-and-cal